src/calendar/calendar-year.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState } from "react";
+import dayjs, { Dayjs } from "dayjs";
+import { YearCell, YearPickerContainer } from "./calendar-year.style";
+import { CalendarHelper } from "../util/calendar-helper";
+import { YearMonthBase } from "./calendar-month";
+
+export type VariantYear =
+    | "default"
+    | "current-year"
+    | "other-decade"
+    | "selected-year";
+
+interface CalendarYearProps extends YearMonthBase {}
```

Comment:
while it seems cleaner, i think it is better to separate the interface for clarity and unnecessary export